In [1]:
library(tidyverse)
library(moderndive)
library(car)

── Attaching core tidyverse packages ─────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ───────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Loading required package: carData


Attaching package: ‘car’


The following object is masked from ‘package:dplyr’:

    recode


The following object is masked from ‘package:purrr’:

    some




In [2]:
wage_data <- read.table("data/wage.txt", header=TRUE)
wage_data$wage_categorical <- as.numeric(wage_data$wage > mean(wage_data$wage))
head(wage_data)

,education,south,sex,experience,union,wage,age,race,occupation,sector,marr,wage_categorical
,<int>,<int>,<int>,<int>,<int>,<dbl>,<int>,<int>,<int>,<int>,<int>,<dbl>
1,8,0,1,21,0,5.10,35,2,6,1,1,0
2,9,0,1,42,0,4.95,57,3,6,1,1,0
3,12,0,0,1,0,6.67,19,3,6,1,0,0
4,12,0,0,4,0,4.00,22,3,6,0,0,0
5,12,0,0,17,0,7.50,35,3,6,0,1,0
6,13,0,0,9,1,13.07,28,3,6,0,0,1


### Part A

In [3]:
wage_logistic_model <- glm(formula = wage_categorical ~ education + sex, data=wage_data, family='binomial')
summary(wage_logistic_model)


Call:
glm(formula = wage_categorical ~ education + sex, family = "binomial", 
    data = wage_data)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept) -4.15150    0.56283  -7.376 1.63e-13 ***
education    0.30942    0.04184   7.395 1.41e-13 ***
sex         -0.81518    0.19492  -4.182 2.89e-05 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 716.62  on 533  degrees of freedom
Residual deviance: 634.26  on 531  degrees of freedom
AIC: 640.26

Number of Fisher Scoring iterations: 3


EDUCATION and SEX are both associated with an individual's wage being above or below average. We can observe `Pr(>|z|)` which indicates the p-value is `1.41e-13` for EDUCATION and `2.89e-05` for SEX. Both of these values are significantly below 0.05 which indicates that the two coefficiencts are statistically significant and therefore there is enough evidence to claim that the coefficients are not 0.

Increased EDUCATION is associated with higher odds of above average wage (`0.30942`). Being female is associated with lower odds of having an above average wage (`-0.81518`) and therefore being a male is associated with higher odds of having an above average wage.

Note: for this dataset, 0=male and 1=female.

### Part B

In [4]:
l_i <- (-4.15150) + (-0.81518) + 16*(0.30942)
estimated_prob <- exp(l_i)/(1+exp(l_i))
estimated_prob

[1] 0.4960101

There is a `49.60 %` probability that a female with 16 years of education has a wage that is above average.